### ISSUE TITLE
Creating namespace fails with "invalid apiVersion client.authentication.k8s.io/v1alpha1" in provider 2.17

### ISSUE THREAD

**Issue by dmacbride-ep** [hashicorp/terraform-provider-kubernetes#1969]
### Terraform Version, Provider Version and Kubernetes Version

```
Terraform version: 0.13
Kubernetes provider version: 2.17.0
Kubernetes version: 1.22 (Amazon EKS)
```

### Affected Resource(s)

- kubernetes_namespace

### Terraform Configuration Files
not available

### Debug Output
not available

### Panic Output
not available

### Steps to Reproduce
I haven't yet had the time to narrow this down to a simple test case. Below are the broad strokes of what we are experiencing.
- Provision an Amazon EKS 1.22 cluster
- Use Terraform 0.13.5 and Kubernetes provider 2.17
- Try to create a 'kubernetes_namespace' resource

### Expected Behavior
The 'kubernetes_namespace' resource is successfully created.

### Actual Behavior
The resource fails to create, with error `Failed to configure client: exec plugin: invalid apiVersion "client.authentication.k8s.io/v1alpha1"`

Showing the error in more context (which includes some statements from our wrapper script):

```
kubernetes-bootstrap    | Acquiring state lock. This may take a few moments...
kubernetes-bootstrap    | kubernetes_namespace.modsec-namespace[0]: Creating...
kubernetes-bootstrap    | 
kubernetes-bootstrap    | Error: Failed to configure client: exec plugin: invalid apiVersion "client.authentication.k8s.io/v1alpha1"
kubernetes-bootstrap    | 
kubernetes-bootstrap    |   on modsec-waf.tf line 5, in resource "kubernetes_namespace" "modsec-namespace":
kubernetes-bootstrap    |    5: resource "kubernetes_namespace" "modsec-namespace" {
kubernetes-bootstrap    | 
kubernetes-bootstrap    | 
kubernetes-bootstrap    | Releasing state lock. This may take a few moments...
kubernetes-bootstrap    | ERROR: failed to create Terraform based resources (for the WAF resources)
kubernetes-bootstrap exited with code 1
```

### Important Factoids
This seems to be an issue in the new Kubernetes provider version 2.17. If I revert to 2.16.1 then the issue goes away.

### References
I could not find any other related GitHub issues.

### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

**Comment by imriz** [comment]
@sebas-w You're using the exec plugin to get the token by any chance?
Something like
```
    exec {
      api_version = "client.authentication.k8s.io/v1alpha1"
      args        = ["eks", "get-token", "--cluster-name", "mycluster"]
      command     = "aws"
    }
```

If so, change the `api_version` to `client.authentication.k8s.io/v1alpha1`

**Comment by imriz** [comment]
This probably comes from https://github.com/hashicorp/terraform-provider-kubernetes/pull/1939 which upgraded the dependencies (this is not in the changelog...).
This upgraded the k8s client which introduced this change:
https://github.com/kubernetes/kubernetes/pull/108616

**Comment by wmedlar** [comment]
Is Kubernetes 1.24 now the minimum version supported by the provider? 1.23 is still [supported until Feb 28](https://kubernetes.io/releases/patch-releases/#1-23), and [EKS 1.23 through October](https://docs.aws.amazon.com/eks/latest/userguide/kubernetes-versions.html#kubernetes-release-calendar).

**Comment by nisargtk** [comment]
I had similar issue. We were using aws-iam-authenticator as exec plugin for kubernetes provider which was having 0.5.0 version which uses client.authentication.k8s.io/v1alpha1 , We upgraded aws-iam-authenticator to 0.6.2 (0.5.4 is also fine) and started using client.authentication.k8s.io/v1beta1 api in kubernetes provider configuration. Also Work around is to hardcode kubernetes provider version to 2.16.1.

**Comment by alexsomesan** [comment]
Are you unable to upgrade the version of the exec plugin to one that supports `client.authentication.k8s.io/v1beta1` or `client.authentication.k8s.io/v1`? 

This should just be a matter of updating the exec plugin itself and should not be tied to cluster version.

**Comment by imriz** [comment]
> Are you unable to upgrade the version of the exec plugin to one that supports `client.authentication.k8s.io/v1beta1` or `client.authentication.k8s.io/v1`? 
> 
> This should just be a matter of updating the exec plugin itself and should not be tied to cluster version.

The problem, at least in my case, was that we were using some 3rd party image, which was using AWS CLI 1.22.x which uses the alpha version, and does not support KUBERNETES_EXEC_INFO either. 

Another point is that none of this is on the change log or upgrade guide.

**Comment by BBBmau** [comment]
Hello! Thank you for opening this issue @dmacbride-ep 

It looks like after reviewing the cli documentation that you can move from alpha to beta from the latest [version](https://awscli.amazonaws.com/v2/documentation/api/latest/reference/eks/get-token.html)

**Comment by simtcrom** [comment]
I am getting this exact error when trying to create simple kubernetes role (kubernetes_role). Error is, 

_│ Error: Failed to configure client: exec plugin: invalid apiVersion "client.authentication.k8s.io/v1alpha1"
│
│   with kubernetes_role.example,
│   on main.tf line 1, in resource "kubernetes_role" "example":
│    1: resource "kubernetes_role" "example" {
│_

If I explicitly use provider version = "2.16.1", everything works. Anything above 2.16.1, I run in to above issue.
I already verified that, aws-iam-authenticator version 0.5.9.
kubeconfig is generated for the cluster with [kconnect](https://github.com/fidelity/kconnect)

Should I be setting up, exec {} under **provider "kubernetes" {}** block?
If so, what should I be giving there?

**Comment by simtcrom** [comment]
> 

I resolved the issue by upgrading kconnect to 0.5.9 and aws-iam-authenticator version to 0.5.9.
Didn't have to put any exec {} under provider configuration.
